packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -145,11 +155,14 @@ export class MongoLogManager {
       // To handle such scenarios, we will catch lstat errors and retry cleaning up
       // to let different processes reach out to different log files.
       if (statErr.code === 'ENOENT' && remainingRetries > 0) {
-        await this.cleanupOldLogFiles(maxDurationMs - (Date.now() - deletionStartTimestamp), remainingRetries - 1);
+        await this.cleanupOldLogFiles(
+          maxDurationMs - (Date.now() - deletionStartTimestamp),
+          remainingRetries - 1
+        );
       }
     }
 
-    if (this._options.retentionGB) {
+    if (hasRetentionGB && this._options.retentionGB) {
```

Comment:
I realize this is redundant but TypeScript isn't smart enough to do type promotion otherwise